parts/k8s/kubernetesconfigscustomcloud.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+AZURE_STACK_ENV="azurestackcloud"
+function ensureCertificates() {
+    if [[ "${TARGET_ENVIRONMENT,,}" == "${AZURE_STACK_ENV}" ]]; then 
+        AZURESTACK_ENVIRONMENT_JSON_PATH="/etc/kubernetes/azurestackcloud.json"
+        AZURESTACK_RESOURCE_MANAGER_ENDPOINT=`cat $AZURESTACK_ENVIRONMENT_JSON_PATH | jq .resourceManagerEndpoint | tr -d "\""`
+        AZURESTACK_RESOURCE_METADATA_ENDPOINT="$AZURESTACK_RESOURCE_MANAGER_ENDPOINT/metadata/endpoints?api-version=2015-01-01"
+        curl $AZURESTACK_RESOURCE_METADATA_ENDPOINT
+        CURL_RETURNCODE=$?
+        if [ $CURL_RETURNCODE != 0 ]; then
+            # Replace placeholder for ssl binding
+            KUBE_CONTROLLER_MANAGER_FILE=/etc/kubernetes/manifests/kube-controller-manager.yaml
+            sed -i "s|<volumessl>|- name: ssl\n      hostPath:\n        path: \\/etc\\/ssl\\/certs|g" $KUBE_CONTROLLER_MANAGER_FILE 
+            sed -i "s|<volumeMountssl>|- name: "ssl"\n          mountPath: \\/etc\\/ssl\\/certs\n          readOnly: true|g" $KUBE_CONTROLLER_MANAGER_FILE
+
+            # Copying the AzureStack root certificate to the appropriate store to be updated.
+            AZURESTACK_ROOT_CERTIFICATE_SOURCE_PATH="/var/lib/waagent/Certificates.pem"
+            AZURESTACK_ROOT_CERTIFICATE__DEST_PATH="/usr/local/share/ca-certificates/azsCertificate.crt"
+            cp $AZURESTACK_ROOT_CERTIFICATE_SOURCE_PATH $AZURESTACK_ROOT_CERTIFICATE__DEST_PATH
+            update-ca-certificates
+        else
+            # the ARM resource manager endpoint binding certificate is trusted, remove the placeholder for ssl binding
+            sed -i "/<volumessl>/d" $KUBE_CONTROLLER_MANAGER_FILE
+            sed -i "/<volumeMountssl>/d" $KUBE_CONTROLLER_MANAGER_FILE
+        fi
+
+        # ensureCertificates will be retried if the exit code is not 0
+        curl $AZURESTACK_RESOURCE_METADATA_ENDPOINT
+        exit $?
+    fi
+}
+
+configureK8sCustomCloud() {
+    if [[ "${TARGET_ENVIRONMENT,,}" == "${AZURE_STACK_ENV}"  ]]; then 
```

Comment:
We don't need this `if` check, as the same `if` check is around the function invocation (where it should be).